### PR TITLE
Add telemetry event in StorageHelper

### PR DESCRIPTION
**Pull request by rpdome**
to see how often this is invoked before https://github.com/AzureAD/ad-accounts-for-android/pull/1196 is done. (Ideally, none).

### CODE REVIEW

File: common/src/main/java/com/microsoft/identity/common/adal/internal/cache/StorageHelper.java [hunk 1]
```diff
@@ -287,6 +287,20 @@ public String decrypt(final String encryptedBlob) throws GeneralSecurityExceptio
...
 
+        // Try to read keystore key - to verify how often this is invoked before the migration is done.
+        // TODO: remove this whole try-catch clause once the experiment is done.
+        if (mTelemetryCallback != null) {
+            try {
+                final SecretKey key = loadSecretKey(KeyType.KEYSTORE_ENCRYPTED_KEY);
+                if (key == null) {
+                    mTelemetryCallback.logEvent(mContext, methodName, false, "KEY_DECRYPTION_KEYSTORE_KEY_NOT_INITIALIZED");
+                }
+            } catch (Exception e) {
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/adal/internal/cache/StorageHelper.java]: nit: `final`

File: common/src/main/java/com/microsoft/identity/common/adal/internal/cache/StorageHelper.java [hunk 2]
```diff
@@ -287,6 +287,20 @@ public String decrypt(final String encryptedBlob) throws GeneralSecurityExceptio
...
+                // Best effort.
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/adal/internal/cache/StorageHelper.java]: In addition to telemetry, should we log here?

File: common/src/main/java/com/microsoft/identity/common/adal/internal/cache/StorageHelper.java [hunk 3]
```diff
@@ -501,12 +515,18 @@ public synchronized SecretKey loadSecretKeyForEncryption() throws IOException,
...
-                // Best effort.
+            // Try to read keystore key - to verify how often this is invoked before the migration is done.
+            // TODO: remove this whole try-catch clause once the experiment is done.
+            if (mTelemetryCallback != null) {
+                try {
+                    final SecretKey key = loadSecretKey(KeyType.KEYSTORE_ENCRYPTED_KEY);
+                    if (key == null) {
+                        mTelemetryCallback.logEvent(mContext, methodName, false, "KEY_ENCRYPTION_KEYSTORE_KEY_NOT_INITIALIZED");
+                    }
+                } catch (Exception e) {
```
Comment (iambmelt) [inline common/src/main/java/com/microsoft/identity/common/adal/internal/cache/StorageHelper.java]: Same as above, nit on `final` + optional log message

File: common/src/main/java/com/microsoft/identity/common/adal/internal/cache/StorageHelper.java (hunk 2 above)
Comment (rpdome) [reply common/src/main/java/com/microsoft/identity/common/adal/internal/cache/StorageHelper.java]: I think we don't need to, as this won't make much sense in log file.

**Comment by rpdome** [comment]
Thanks @iambmelt @shahzaibj !

**Pull request merged**